package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/SceneImporter.cs

```diff
@@ -309,6 +319,16 @@ public ImportException(string message, System.Exception innerException)
       // A PrimMap is returned for tracking the USD <-> Unity mapping.
       Profiler.BeginSample("USD: Build Hierarchy");
       if (importOptions.importHierarchy || importOptions.forceRebuild) {
+        // TODO(jcowles): This feels like a workaround. What we really want here is an "undo"
+        // process for changes made to the root GameObject. For example, to clean up non-USD
+        // components which may have been added.
+        RemoveComponent<UsdAssemblyRoot>(root);
+        RemoveComponent<UsdVariantSet>(root);
+        RemoveComponent<UsdModelRoot>(root);
+        RemoveComponent<UsdLayerStack>(root);
+        RemoveComponent<UsdPayload>(root);
+        RemoveComponent<UsdPrimSource>(root);
```

Comment:
I'm not sure why you added all this `RemoveComponent` thing and how it is linked to the original problem?

Reply:
When a USD file is fully RE-imported, all exsiting USD data must be removed. The old assumption was that the root would never have much more than the UsdAsset component itself, however it's now clear that the root may also have meaningful USD data added too.

This is an attempt to clean the root UsdAsset GameObject on re-import, however as I mentioned in the TODO comment, it it is incomplete (imagine if a mesh is imported to the root) and we really need a full "undo" operation to clean the root GameObject. Unfortunately, this is very difficult currently.

I'll add this comment in the code as well!